### PR TITLE
Add prop to disable the visibility check

### PR DESCRIPTION
**Pull request by wolnewitz**
- react visiblity sensor needed to be updated to play nicely with React
16, it was still using `React.createClass`

- Added a option to turn this optimization off for users who are facing
rendering issues with it in React 16+.

### CODE REVIEW

File: src/components/drivers/pdf-viewer.jsx [hunk 1]
```diff
@@ -9,15 +9,23 @@ PDFJS.disableWorker = true;
...
-
+export class PDFPage extends React.Component {
   constructor(props) {
     super(props);
     this.state = {};
     this.onChange = this.onChange.bind(this);
   }
 
+  componentDidMount() {
+    if (this.props.disableVisibilityCheck) this.fetchAndRenderPage();
```
Comment (aaronnorby) [inline src/components/drivers/pdf-viewer.jsx]: why is this necessary?

File: src/components/fetch-wrapper.jsx [hunk 2]
```diff
@@ -86,6 +69,23 @@ function withFetching(WrappedComponent, props) {
         this.xhr.abort();
       }
     }
+
+    render() {
```
Comment (aaronnorby) [inline src/components/fetch-wrapper.jsx]: what's the reason for the change in this file?

File: src/components/drivers/pdf-viewer.jsx [hunk 3]
```diff
@@ -9,15 +9,23 @@ PDFJS.disableWorker = true;
 const INCREASE_PERCENTAGE = 0.2;
 const DEFAULT_SCALE = 1.1;
 
-class PDFPage extends React.Component {
-
+export class PDFPage extends React.Component {
```
Comment (aaronnorby) [inline src/components/drivers/pdf-viewer.jsx]: can this component be used on its own? why export it?

Comment (wolnewitz) [reply src/components/drivers/pdf-viewer.jsx]: Yes it can be moved to a different file if you want, I export it because I wrote a few few tests for it.  But I didn't move it so that the code changes could be seen more clearly.

File: src/components/drivers/pdf-viewer.jsx (hunk 1 above)
Comment (wolnewitz) [reply src/components/drivers/pdf-viewer.jsx]: The pdf was getting rendered before only in `componentDidUpdate` based of that visibility check, so now we can/have to render the pdf into the DOM right away.

File: src/components/fetch-wrapper.jsx (hunk 2 above)
Comment (wolnewitz) [reply src/components/fetch-wrapper.jsx]: This was a lint error causing `npm start` to fail, I have removed this change and make a different PR for it.

**Comment by wolnewitz** [comment]
Also this didn't have a `package-lock` file checked in, I added it in this PR but let me know if you think that doesn't belong in this PR. @aaronnorby

File: src/components/drivers/pdf-viewer.jsx (hunk 3 above)
Comment (aaronnorby) [reply src/components/drivers/pdf-viewer.jsx]: cool - no it's ok to leave it here
